lib/fetch/request.js

```diff
@@ -728,7 +728,7 @@ class Request {
     const clonedRequestObject = new Request(kInit)
     clonedRequestObject[kState] = clonedRequest
     clonedRequestObject[kRealm] = this[kRealm]
-    clonedRequestObject[kHeaders] = new Headers()
+    clonedRequestObject[kHeaders] = new Headers(kConstruct)
```

Comment:
```suggestion
    clonedRequestObject[kHeaders] = Object.create(Headers.prototype)
```

Wouldn't this achieve the same?

Reply:
It is faster to use the new operator.

Reply:
```js
const benchmark = require("benchmark");

const suite = new benchmark.Suite();

// bench.js
const kSkip = Symbol("skip");

class FakeClass {
  constructor(skip) {
    if (skip !== kSkip) throw new TypeError("Illegal constructor");
  }
}

suite.add("new operator", () => {
  new FakeClass(kSkip);
});

suite.add("Object.create", () => {
  Object.create(FakeClass.prototype);
});

suite.on("cycle", function (event) {
  console.log(String(event.target));
});

// run async
new Promise((resolve) => setTimeout(resolve, 7000)).then(() =>
  suite.run({ async: true })
);
```

```
new operator x 720,079,182 ops/sec ±1.36% (83 runs sampled)
Object.create x 710,966,585 ops/sec ±1.43% (85 runs sampled)
```

Reply:
The difference is within the margin of error. I would prefer Object.create since it's less "hacky".